crates/rpc/src/method/get_messages_status.rs

```diff
@@ -0,0 +1,119 @@
+use anyhow::Context;
+use pathfinder_common::{ChainId, EthereumChain, L1TransactionHash, TransactionHash};
+use pathfinder_ethereum::EthereumApi;
+use serde::{Deserialize, Serialize};
+
+use crate::context::RpcContext;
+use crate::method::get_transaction_status;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct Input {
+    transaction_hash: L1TransactionHash,
+}
+
+impl crate::dto::DeserializeForVersion for Input {
+    fn deserialize(value: crate::dto::Value) -> Result<Self, serde_json::Error> {
+        value.deserialize_map(|value| {
+            Ok(Self {
+                transaction_hash: value
+                    .deserialize("transaction_hash")
+                    .map(L1TransactionHash::new)?,
+            })
+        })
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+enum FinalityStatus {
+    Received,
+    Rejected,
+    AcceptedOnL2,
+    AcceptedOnL1,
+}
+
+#[derive(Clone, Debug)]
+pub struct L1HandlerTransactionStatus {
+    transaction_hash: TransactionHash,
+    finality_status: FinalityStatus,
+    failure_reason: Option<String>,
+}
+
+#[derive(Debug)]
+pub struct Output(Vec<L1HandlerTransactionStatus>);
+
+crate::error::generate_rpc_error_subset!(Error: TxnHashNotFound);
+
+pub async fn get_messages_status(context: RpcContext, input: Input) -> Result<Output, Error> {
+    let span = tracing::Span::current();
+
+    let _g = span.enter();
+
+    // Fetch the L1 handler transactions for the given transaction hash
+    let ethereum = context.ethereum.clone();
+    let eth_chain = ethereum.get_chain().await?;
+    let l1_handler_txs = ethereum
+        .get_l1_handler_txs(eth_chain, &input.transaction_hash)
+        .await
+        .context("Fetching L1 handler tx hashes")
+        .map_err(|_| Error::TxnHashNotFound)?;
+
+    let strk_chain = match eth_chain {
+        EthereumChain::Mainnet => ChainId::MAINNET,
+        EthereumChain::Sepolia => ChainId::SEPOLIA_TESTNET,
+        _ => return Err(anyhow::anyhow!("Unsupported chain").into()),
+    };
```

Comment:
We should just use `context.chain_id` instead.

Reply:
Indeed